packages/codemirror/src/codemirror.js

```diff
@@ -300,43 +303,69 @@ export function createCypherEditor(parentDOMElement, options = {}) {
     fire: fireKeyDown
   } = createEventHandlers();
 
+  const { on: onKeyUp, off: offKeyUp, fire: fireKeyUp } = createEventHandlers();
+
   const lineNumberClick = (line, event) => {
     fireLineNumberClick(line, event);
   };
 
   const keyDown = (event) => {
+    pressedKey =
+      !event.metaKey &&
+      !event.ctrlKey &&
+      !event.altKey &&
+      event.key &&
+      event.key.length === 1
+        ? event.key
+        : null;
     fireKeyDown(event);
   };
 
-  const handleValueChanged = (value, changes) => {
+  const keyUp = (event) => {
+    if (deferredAutocomplete) {
+      showAutocomplete();
+      deferredAutocomplete = false;
+    }
+
+    pressedKey = null;
+    fireKeyUp(event);
+  };
+
+  const deferAutocompleteForChanges = (changes) => {
+    const { length = 0, newLength = 0 } = changes;
+    const isSmallInsert = newLength > length && newLength - length <= 2;
     if (
+      pressedKey &&
       cypherLanguage &&
       autocomplete &&
-      Array.isArray(autocompleteTriggerStrings)
+      Array.isArray(autocompleteTriggerStrings) &&
+      isSmallInsert
     ) {
-      let changedText = [];
+      let changedText = null;
       changes.iterChanges((fromA, toA, fromB, toB, inserted) => {
-        // If changes have more than 32 lines, inserted will have a children property
-        // with 32 lines in each child.
-        if (inserted.children) {
-          changedText = inserted.children.map((c) => c.text);
-          return;
+        if (
+          !changedText &&
+          inserted &&
+          typeof inserted === "object" &&
+          Array.isArray(inserted.text) &&
+          inserted.text.length == 1
+        ) {
+          changedText = inserted.text[0];
+          // { {}
         }
-        changedText = inserted.text;
       });
 
-      if (changedText.length > 0 && changedText.length <= 2) {
-        const text = changedText[0];
-        if (autocompleteTriggerStrings.indexOf(text) !== -1) {
-          showDeferredAutocomplete();
-        } else if (changedText.length === 2) {
-          const longerText = text + changedText[1];
-          if (autocompleteTriggerStrings.indexOf(longerText) !== -1) {
-            showDeferredAutocomplete();
-          }
-        }
+      if (
+        changedText &&
+        autocompleteTriggerStrings.indexOf(changedText) !== -1
```

Comment:
no need to change it, but is there some configuration reason that this is not an `.includes` ?

Reply:
nope, changed it